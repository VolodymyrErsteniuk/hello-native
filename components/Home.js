import React, { useState } from 'react';
import { SafeAreaView, ScrollView, Text, TouchableOpacity } from 'react-native';
import { Header } from 'react-native/Libraries/NewAppScreen';

function Home() {
  const [greeting, setGreeting] = useState('');

  const sayHi = () => {
    setGreeting('Welcome to SoftServe');
  };

  return (
    <SafeAreaView>
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <Header />

        <TouchableOpacity testID="sayHi" onPress={sayHi}>
          <Text>Say Hi</Text>
        </TouchableOpacity>

        <Text testID="greeting">{greeting}</Text>
      </ScrollView>
    </SafeAreaView>
  );
}

export default Home;
