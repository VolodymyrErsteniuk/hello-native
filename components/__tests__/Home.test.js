import React from 'react';
import {shallow} from 'enzyme';
import Home from '../../components/Home';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Home />);
});

it('renders correctly', () => {
  expect(wrapper).toExist();
});

it('matches the snapshot', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should change greeting by pressing the button', () => {
  const button = wrapper.findWhere(node => node.prop('testID') === 'sayHi');
  button.simulate('press');
  wrapper.update();

  const greeting = wrapper.findWhere(node => node.prop('testID') === 'greeting');

  expect(greeting.props().children).toBe('Welcome to SoftServe');
});
