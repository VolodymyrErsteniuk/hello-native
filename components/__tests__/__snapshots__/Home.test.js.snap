// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<SafeAreaView>
  <ScrollViewMock
    contentInsetAdjustmentBehavior="automatic"
  >
    <Header />
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      testID="sayHi"
    >
      <Text>
        Say Hi
      </Text>
    </TouchableOpacity>
    <Text
      testID="greeting"
    />
  </ScrollViewMock>
</SafeAreaView>
`;
